Test failures on Windows

Hello,

Could you please have a look on this error. I am building a fresh cloned source. Thank you.

68 tests completed, 35 failed
:org.eclipse.xtext.web:test FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':org.eclipse.xtext.web:test'.
> There were failing tests. See the report at: file:///C:/Users/****/Git/xtext-web3/org.eclipse.xtext.web/build/reports/tests/test/index.html

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 2 mins 42.843 secs

--- a/org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/AbstractWebServerTest.xtend
+++ b/org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/AbstractWebServerTest.xtend
@@ -15,6 +15,7 @@ import java.io.FileWriter
 import java.util.HashMap
 import java.util.Map
 import org.eclipse.emf.common.util.URI
+import org.eclipse.xtext.formatting.ILineSeparatorInformation
 import org.eclipse.xtext.util.DisposableRegistry
 import org.eclipse.xtext.util.Modules2
 import org.eclipse.xtext.web.example.statemachine.StatemachineRuntimeModule
@@ -26,9 +27,26 @@ import org.eclipse.xtext.web.server.XtextServiceDispatcher
 import org.eclipse.xtext.web.server.persistence.IResourceBaseProvider
 import org.eclipse.xtext.web.server.test.languages.StatemachineWebModule
 import org.junit.After
+import org.junit.Assert
 import org.junit.Before
 
 abstract class AbstractWebServerTest {
+	
+	static class TestLineSeparatorInformation implements ILineSeparatorInformation {
+		
+		override getLineSeparator() {
+			"\n"
+		}
+		
+	}
+	
+	protected def normalizeLineSeparator(String value) {
+		value.replace("\r\n","\n")
+	}
+
+	def public void assertEquals(CharSequence expected, CharSequence actual) {
+		Assert.assertEquals(expected.toString.normalizeLineSeparator, actual.toString.toString.normalizeLineSeparator)
+	}
 	
 	static class TestResourceBaseProvider implements IResourceBaseProvider {
 		val testFiles = new HashMap<String, URI>
@@ -42,8 +60,7 @@ abstract class AbstractWebServerTest {
 		override protected internalCreateInjector() {
 			new StatemachineStandaloneSetup {
 				override createInjector() {
-					val ideModule = new StatemachineIdeModule
-					val webModule = new StatemachineWebModule
+					val webModule = getWebModule()
 					webModule.resourceBaseProvider = resourceBaseProvider
 					return Guice.createInjector(Modules2.mixin(runtimeModule, ideModule, webModule))
 				}
@@ -57,8 +74,20 @@ abstract class AbstractWebServerTest {
 	
 	@Inject XtextServiceDispatcher dispatcher
 	
+	protected def Module getIdeModule() {
+		new StatemachineIdeModule
+	}
+	
+	protected def StatemachineWebModule getWebModule() {
+		new StatemachineWebModule
+	}
+	
 	protected def Module getRuntimeModule() {
-		new StatemachineRuntimeModule
+		new StatemachineRuntimeModule() {
+			def Class<? extends ILineSeparatorInformation> bindILineSeparatorInformation() {
+				TestLineSeparatorInformation
+			}
+		}
 	}
 	
 	@Before

--- a/org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/Bug489571Test.xtend
+++ b/org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/Bug489571Test.xtend
@@ -1,9 +1,10 @@
 package org.eclipse.xtext.web.server.test
 
 import org.eclipse.xtext.formatting.IIndentationInformation
+import org.eclipse.xtext.formatting.ILineSeparatorInformation
 import org.eclipse.xtext.web.example.statemachine.StatemachineRuntimeModule
 import org.eclipse.xtext.web.server.formatting.FormattingResult
-import org.eclipse.xtext.web.server.test.Bug489571Test.Bug489571IndentationInformation
+import org.eclipse.xtext.web.server.test.AbstractWebServerTest.TestLineSeparatorInformation
 import org.junit.Test
 
 import static org.junit.Assert.*
@@ -15,6 +16,10 @@ class Bug489571Test extends AbstractWebServerTest {
 			def Class<? extends IIndentationInformation> bindIIndentationInformation() {
 				Bug489571IndentationInformation
 			}
+			def Class<? extends ILineSeparatorInformation> bindILineSeparatorInformation() {
+				TestLineSeparatorInformation
+			}
+			
 		}
 	}
 

--- a/org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/ContentAssistTest.xtend
+++ b/org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/ContentAssistTest.xtend
@@ -7,6 +7,8 @@
  *******************************************************************************/
 package org.eclipse.xtext.web.server.test
 
+import org.eclipse.xtext.web.example.statemachine.ide.StatemachineIdeModule
+import org.eclipse.xtext.web.example.statemachine.ide.StatemachineTemplateProposalProvider
 import org.eclipse.xtext.web.server.ServiceConflictResult
 import org.eclipse.xtext.web.server.contentassist.ContentAssistResult
 import org.junit.Test
@@ -15,6 +17,23 @@ import static org.hamcrest.core.IsInstanceOf.*
 import static org.junit.Assert.*
 
 class ContentAssistTest extends AbstractWebServerTest {
+	
+	override protected getIdeModule() {
+		new StatemachineIdeModule() {
+				def Class<? extends StatemachineTemplateProposalProvider> bindStatemachineTemplateProposalProvider() {
+				StatemachineTemplateProposalProvider2
+			}
+
+		}
+	}
+	
+	static class StatemachineTemplateProposalProvider2 extends StatemachineTemplateProposalProvider {
+		
+		override protected getLineDelimiter() {
+			"\n"
+		}
+		
+	}
 	
 	protected def assertContentAssistResult(CharSequence resourceContent, CharSequence expectedResult) {
 		var contentString = resourceContent.toString
@@ -120,7 +139,7 @@ class ContentAssistTest extends AbstractWebServerTest {
 		'''
 			state foo end
 				|
-		'''.assertContentAssistResult('''
+		'''.toString.normalizeLineSeparator.assertContentAssistResult('''
 			ContentAssistResult [
 			  stateId = "-80000000"
 			  entries = ArrayList (

--- a/org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/Formatting1Test.xtend
+++ b/org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/Formatting1Test.xtend
@@ -7,8 +7,10 @@
  *******************************************************************************/
 package org.eclipse.xtext.web.server.test
 
+import org.eclipse.xtext.formatting.ILineSeparatorInformation
 import org.eclipse.xtext.web.example.statemachine.StatemachineRuntimeModule
 import org.eclipse.xtext.web.server.formatting.FormattingResult
+import org.eclipse.xtext.web.server.test.AbstractWebServerTest.TestLineSeparatorInformation
 import org.eclipse.xtext.web.server.test.languages.formatting.StatemachineFormatter
 import org.junit.Test
 
@@ -24,9 +26,12 @@ class Formatting1Test extends AbstractWebServerTest {
 			override bindIFormatter() {
 				StatemachineFormatter
 			}
+			def Class<? extends ILineSeparatorInformation> bindILineSeparatorInformation() {
+				TestLineSeparatorInformation
+			}
 		}
 	}
-	
+
 	@Test def testFormatFile() {
 		val file = createFile('/* bla */ output signal x state foo set x = true end')
 		val format = getService(#{'serviceType' -> 'format', 'resource' -> file.name})

--- a/org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/GeneratorTest.xtend
+++ b/org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/GeneratorTest.xtend
@@ -8,6 +8,7 @@
 package org.eclipse.xtext.web.server.test
 
 import org.eclipse.emf.ecore.resource.Resource
+import org.eclipse.xtext.formatting.ILineSeparatorInformation
 import org.eclipse.xtext.generator.AbstractGenerator
 import org.eclipse.xtext.generator.IFileSystemAccess2
 import org.eclipse.xtext.generator.IGeneratorContext
@@ -15,7 +16,7 @@ import org.eclipse.xtext.web.example.statemachine.StatemachineRuntimeModule
 import org.eclipse.xtext.web.example.statemachine.statemachine.Statemachine
 import org.eclipse.xtext.web.server.generator.GeneratorResult
 import org.eclipse.xtext.web.server.generator.GeneratorService
-import org.eclipse.xtext.web.server.test.GeneratorTest.Generator
+import org.eclipse.xtext.web.server.test.AbstractWebServerTest.TestLineSeparatorInformation
 import org.junit.Test
 
 import static org.junit.Assert.*
@@ -47,6 +48,9 @@ class GeneratorTest extends AbstractWebServerTest {
 		new StatemachineRuntimeModule {
 			override bindIGenerator2() {
 				Generator
+			}
+			def Class<? extends ILineSeparatorInformation> bindILineSeparatorInformation() {
+				TestLineSeparatorInformation
 			}
 		}
 	}

--- a/org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/OccurrenceTest.xtend
+++ b/org.eclipse.xtext.web/src/test/java/org/eclipse/xtext/web/server/test/OccurrenceTest.xtend
@@ -26,7 +26,7 @@ class OccurrenceTest extends AbstractWebServerTest {
 	}
 	
 	def protected assertOccurrences(CharSequence resourceContent, CharSequence expectation) {
-		assertEquals(expectation.toString.trim, resourceContent.occurrences.toString.trim)
+		assertEquals(expectation.toString.trim, resourceContent.toString.normalizeLineSeparator.occurrences.toString.trim)
 	}
 	
 	@Test 

--- a/org.eclipse.xtext.xbase.web/src/test/java/org/eclipse/xtext/xbase/web/test/AbstractXbaseWebTest.xtend
+++ b/org.eclipse.xtext.xbase.web/src/test/java/org/eclipse/xtext/xbase/web/test/AbstractXbaseWebTest.xtend
@@ -27,6 +27,7 @@ import org.eclipse.xtext.web.server.persistence.IResourceBaseProvider
 import org.eclipse.xtext.xbase.web.test.languages.EntitiesWebModule
 import org.junit.After
 import org.junit.Before
+import org.junit.Assert
 
 abstract class AbstractXbaseWebTest {
 	
@@ -36,6 +37,14 @@ abstract class AbstractXbaseWebTest {
 		override getFileURI(String resourceId) {
 			testFiles.get(resourceId)
 		}
+	}
+	
+	protected def normalizeLineSeparator(String value) {
+		value.replace("\r\n","\n")
+	}
+	
+	def public void assertEquals(CharSequence expected, CharSequence actual) {
+		Assert.assertEquals(expected.toString.normalizeLineSeparator, actual.toString.toString.normalizeLineSeparator)
 	}
 	
 	val injectorProvider = new EntitiesInjectorProvider {

--- a/org.eclipse.xtext.xbase.web/src/test/java/org/eclipse/xtext/xbase/web/test/ContentAssistTest.xtend
+++ b/org.eclipse.xtext.xbase.web/src/test/java/org/eclipse/xtext/xbase/web/test/ContentAssistTest.xtend
@@ -10,12 +10,10 @@ package org.eclipse.xtext.xbase.web.test
 import org.eclipse.xtext.web.server.contentassist.ContentAssistResult
 import org.junit.Test
 
-import static org.junit.Assert.*
-
 class ContentAssistTest extends AbstractXbaseWebTest {
 	
 	protected def assertContentAssistResult(CharSequence resourceContent, CharSequence expectedResult) {
-		var contentString = resourceContent.toString
+		var contentString = resourceContent.toString.normalizeLineSeparator
 		val cursorOffset = contentString.indexOf('|')
 		if (cursorOffset >= 0) {
 			contentString = contentString.substring(0, cursorOffset) + contentString.substring(cursorOffset + 1)

